src/main/java/com/walmartlabs/x12/standard/StandardX12Parser.java

```diff
@@ -163,6 +167,26 @@ public void registerUnhandledTransactionSet(UnhandledTransactionSet txUnhandled)
         this.unhandledTransactionSet = txUnhandled;
     }
 
+    /**
+     * All EDI X12 messages MUST be enclosed in a ISA/ISE envelope.
+     *
+     * @param segmentList
+     * @return true when valid and false otherwise
+     */
+    private boolean isValidEnvelope(List<X12Segment> segmentList) {
+        boolean isValidEnvelope = false;
+        int lastSegmentIndex = segmentList.size() - 1;
```

Comment:
`segmentList.size()` can be assigned to a variable, as it is being used 2 times

Reply:
also is it possible `segmentList` can be empty, in that case need to check `! empty()` 
before `segmentList.size() - 1`

Reply:
unlikely it will be empty, given private method and final class, but also doesn't matter if it is empty
```
int segmentCount = segmentList.size();       // empty so gives 0
int lastSegmentIndex = segmentCount - 1; // 0-1 = -1
if (segmentCount > 1) {                                 // -1 > 1 = false so correctly yields invalid envelope
```